src/System.CommandLine/Binding/ServiceProviderValueSource.cs

```diff
@@ -7,10 +7,10 @@ internal class ServiceProviderValueSource : IValueSource
     {
         public bool TryGetValue(
             IValueDescriptor valueDescriptor,
-            BindingContext bindingContext,
-            out object boundValue)
+            BindingContext? bindingContext,
+            out object? boundValue)
         {
-            boundValue = bindingContext.ServiceProvider.GetService(valueDescriptor.ValueType);
+            boundValue = bindingContext?.ServiceProvider.GetService(valueDescriptor.ValueType);
```

Comment:
I don't know about this repo much. Should logic changes come with a corresponding test?